1-JavaScript/Week4/homework/ex2-mondaysWorth.js

```diff
@@ -29,16 +29,37 @@ const mondayTasks = [
 
 const hourlyRate = 25;
 
-function computeEarnings(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+function computeEarnings(arr, rate) {
+  /*Make sure we use the function computeEarnings on the array (or the object) of objects and there is
+  at least one `duration` property with a number value.*/
+  if (
+    (typeof arr === 'object') &
+    arr.every((val) => typeof val === 'object') &
+    arr.some((val) => {
+      return typeof val.duration === 'number';
+    })
+  ) {
+    const cost = function (obj) {
+      if (typeof obj.duration === 'number') {
```

Comment:
In the if statement above I checked if the array has objects and there is  at least one `duration` property with a number value. And here I check if the current property `duration`  has a number value. 
Because it may happen that in an array one object has a property `duration` with a number value,   and another object has a property `duration` with a not-number value. Then the check above will be passed successfully, and this check will be not passed. 
Or am I thinking the wrong way?

Reply:
You are right, since you used a `some` above. Consider my comment not given!